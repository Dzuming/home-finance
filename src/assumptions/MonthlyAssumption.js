import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { setStatusColor } from '../helpers/Status';
import { Card, CardContent, Grid, Typography, withStyles } from 'material-ui';
import { makeGetMonthlyAssumption } from '../helpers/selectors';
import { yearMonthFormatDate } from '../helpers/format';
import { bindActionCreators, compose } from 'redux';
import * as assumptionActions from '../actions/assumptionActions';
import { connect } from 'react-redux';
import DatePicker from '../commons/DatePicker';

const styles = theme => ({
  isOverdraw: {
    color: theme.status.warning,
  },
  assumptionWrapper: {
    padding: '15px 0',
  },
});

class MonthlyAssumption extends Component {
  state = {
    date: '',
  };
  handleDateChange = event => {
    const date = event.target.value;
    this.setState({ date });
  };

  componentDidMount() {
    this.setState({ date: yearMonthFormatDate });
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.date !== this.state.date) {
      this.props.actions.fetchAssumptions(this.state.date);
    }
  }

  render() {
    const { classes, monthlyAssumption } = this.props;
    const { date } = this.state;
    return (
      <React.Fragment>
        <Grid container spacing={0} className={classes.assumptionWrapper}>
          <DatePicker value={date} handleChange={this.handleDateChange} />
        </Grid>
        <Grid container spacing={0}>
          <Grid item xs={3}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  Assumption
                </Typography>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={3}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  percentage
                </Typography>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={3}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  Value
                </Typography>
              </CardContent>
            </Card>
          </Grid>
          <Grid item xs={3}>
            <Card>
              <CardContent>
                <Typography type="body2" gutterBottom align="center">
                  Limit
                </Typography>
              </CardContent>
            </Card>
          </Grid>
        </Grid>
        {monthlyAssumption.map(assumption => (
          <Grid
            key={assumption.id}
            container
            spacing={0}
            className={setStatusColor(
              assumption.limit - assumption.value,
              '',
              classes.isOverdraw,
            )}
          >
            <Grid item xs={3}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.name}
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={3}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.percentage}%
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={3}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.value} zł
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
            <Grid item xs={3}>
              <Card>
                <CardContent>
                  <Typography color="inherit" type="body2" align="center">
                    {assumption.limit} zł
                  </Typography>
                </CardContent>
              </Card>
            </Grid>
          </Grid>
        ))}
      </React.Fragment>
    );
  }
}

MonthlyAssumption.propTypes = {
  actions: PropTypes.shape({
    fetchAssumptions: PropTypes.Func,
  }),
  monthlyAssumption: PropTypes.array.isRequired,
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  monthlyAssumption: makeGetMonthlyAssumption(state),
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(assumptionActions, dispatch),
});

export default compose(
  withStyles(styles),
  connect(mapStateToProps, mapDispatchToProps),
)(MonthlyAssumption);
