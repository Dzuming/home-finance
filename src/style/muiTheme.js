import { createMuiTheme } from 'material-ui/styles';

export const theme = createMuiTheme({
  status: {
    success: '#4CAF50',
    warning: '#F44336',
  },
});
