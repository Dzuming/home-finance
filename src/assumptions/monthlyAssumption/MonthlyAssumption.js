import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { setStatusColor } from '../../helpers/Status';
import { Card, CardContent, Grid, Typography, withStyles } from 'material-ui';
import { makeGetMonthlyAssumption } from '../../selectors/assumptions';
import { yearMonthFormatDate } from '../../helpers/format';
import { bindActionCreators, compose } from 'redux';
import * as assumptionActions from '../../actions/assumptionActions';
import { connect } from 'react-redux';
import DatePicker from '../../components/commons/DatePicker';
import CardList from '../../components/commons/CardList';

const styles = theme => ({
  isOverdraw: {
    color: theme.status.warning,
  },
  assumptionWrapper: {
    padding: '15px 0',
  },
});

class MonthlyAssumption extends Component {
  state = {
    date: '',
  };
  handleDateChange = event => {
    const date = event.target.value;
    this.setState({ date });
  };

  componentDidMount() {
    this.setState({ date: yearMonthFormatDate });
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.date !== this.state.date) {
      this.props.actions.fetchAssumptions(this.state.date);
    }
  }

  render() {
    const { classes, monthlyAssumption } = this.props;
    const { date } = this.state;
    return (
      <React.Fragment>
        <Grid container spacing={0} className={classes.assumptionWrapper}>
          <DatePicker
            value={date}
            handleChange={this.handleDateChange}
            label={'assumption date'}
          />
        </Grid>
        <Grid container spacing={0}>
          <CardList name={'Assumption'} gridSize={3} />
          <CardList name={'Percentage'} gridSize={3} />
          <CardList name={'Value'} gridSize={3} />
          <CardList name={'Limit'} gridSize={3} />
        </Grid>
        {monthlyAssumption.map(assumption => (
          <Grid
            key={assumption.id}
            container
            spacing={0}
            className={setStatusColor(
              assumption.limit - assumption.value,
              '',
              classes.isOverdraw,
            )}
          >
            <CardList name={assumption.name} gridSize={3} />
            <CardList name={assumption.percentage} gridSize={3} />
            <CardList name={`${assumption.value} zł`} gridSize={3} />
            <CardList name={`${assumption.limit} zł`} gridSize={3} />
          </Grid>
        ))}
      </React.Fragment>
    );
  }
}

MonthlyAssumption.propTypes = {
  actions: PropTypes.shape({
    fetchAssumptions: PropTypes.Func,
  }),
  monthlyAssumption: PropTypes.array.isRequired,
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  monthlyAssumption: makeGetMonthlyAssumption(state),
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(assumptionActions, dispatch),
});

export default compose(
  withStyles(styles),
  connect(mapStateToProps, mapDispatchToProps),
)(MonthlyAssumption);
