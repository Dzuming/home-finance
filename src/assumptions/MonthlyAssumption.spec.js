import React from 'react';
import { create } from 'react-test-renderer';
import MonthlyAssumption from './MonthlyAssumption';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { theme } from '../style/muiTheme';
import { MuiThemeProvider } from 'material-ui';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({
  assumptions: {
    monthly: [],
  },
  user: [],
  login: { isAuthenticated: true },
});
const component = create(
  <MuiThemeProvider theme={theme}>
    <Provider store={store}>
      <MonthlyAssumption />
    </Provider>
  </MuiThemeProvider>,
);

describe('<MonthlyAssumption  />', () => {
  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
