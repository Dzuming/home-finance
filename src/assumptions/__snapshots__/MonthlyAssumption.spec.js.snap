// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthlyAssumption  /> should match snapshot 1`] = `
Array [
  <div
    className="MuiGrid-typeContainer-3 Connect-MonthlyAssumption--assumptionWrapper-2"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-93"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-103 MuiInputLabel-root-97 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-shrink-100"
        data-shrink={true}
        htmlFor="month"
      >
        Assumption date
      </label>
      <div
        className="MuiInput-root-107 MuiInput-formControl-108 MuiInput-inkbar-109 MuiInput-underline-113"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-116 MuiInput-inputSingleline-119"
          defaultValue={undefined}
          disabled={false}
          id="month"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="month"
          value="2018-05"
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-typeContainer-3"
  >
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-3-31"
    >
      <div
        className="MuiPaper-root-122 MuiPaper-shadow2-126 MuiPaper-rounded-123"
      >
        <div
          className="MuiCardContent-root-149"
        >
          <aside
            className="MuiTypography-root-150 MuiTypography-body2-158 MuiTypography-gutterBottom-167 MuiTypography-alignCenter-163"
          >
            Assumption
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-3-31"
    >
      <div
        className="MuiPaper-root-122 MuiPaper-shadow2-126 MuiPaper-rounded-123"
      >
        <div
          className="MuiCardContent-root-149"
        >
          <aside
            className="MuiTypography-root-150 MuiTypography-body2-158 MuiTypography-gutterBottom-167 MuiTypography-alignCenter-163"
          >
            percentage
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-3-31"
    >
      <div
        className="MuiPaper-root-122 MuiPaper-shadow2-126 MuiPaper-rounded-123"
      >
        <div
          className="MuiCardContent-root-149"
        >
          <aside
            className="MuiTypography-root-150 MuiTypography-body2-158 MuiTypography-gutterBottom-167 MuiTypography-alignCenter-163"
          >
            Value
          </aside>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-typeItem-4 MuiGrid-grid-xs-3-31"
    >
      <div
        className="MuiPaper-root-122 MuiPaper-shadow2-126 MuiPaper-rounded-123"
      >
        <div
          className="MuiCardContent-root-149"
        >
          <aside
            className="MuiTypography-root-150 MuiTypography-body2-158 MuiTypography-gutterBottom-167 MuiTypography-alignCenter-163"
          >
            Limit
          </aside>
        </div>
      </div>
    </div>
  </div>,
]
`;
